### PR TITLE
update(ci): update ci jobs to generate Falco images with the modern BPF probe

### PR DESCRIPTION
**Pull request by Andreagit97**
**What type of PR is this?**

/kind feature

**Any specific area of the project related to this PR?**

/area build

/area CI

**What this PR does / why we need it**:

This PR brings the necessary changes in the circle-ci jobs to build a Falco image for the modern bpf probe. I added a docker image that returns Falco packages as output (deb, rpm, tar).

This PR is the dual of https://github.com/falcosecurity/falco/pull/2282, the idea is to merge this for the release and leave the other PR open until Falco 0.34 is released so users can continue to test the modern probe if they want :) 

**Which issue(s) this PR fixes**:

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
update(ci): update ci jobs to generate Falco images with the modern BPF probe
```

### CODE REVIEW

**Comment by FedeDP** [comment]
Can you use some comments in the circleci config? :)

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,51 +3,54 @@ jobs:
...
-          name: Run unit tests
+          name: Install deps ⛓️
           command: |
-            docker run -e BUILD_TYPE="release" -it -v /tmp/source-arm64:/source -v /tmp/build-arm64:/build \
-              falcosecurity/falco-builder:latest \
-              tests
+            sudo apt update
+            sudo apt install -y --no-install-recommends ca-certificates cmake build-essential clang-14 git pkg-config autoconf automake libelf-dev
+            sudo update-alternatives --install /usr/bin/clang clang /usr/bin/clang-14 90
+            sudo update-alternatives --install /usr/bin/llvm-strip llvm-strip /usr/bin/llvm-strip-14 90
```
Comment (Andreagit97) [inline .circleci/config.yml:17]: here we set llvm-14 and clang-14 as default in the system

File: .circleci/config.yml [hunk 2]
```diff
@@ -3,51 +3,54 @@ jobs:
...
+            git clone https://github.com/libbpf/bpftool.git --branch v7.0.0 --single-branch
+            cd bpftool
+            git submodule update --init
+            cd src && sudo make install
```
Comment (Andreagit97) [inline .circleci/config.yml:21]: here we install bpftool

File: .circleci/config.yml [hunk 3]
```diff
@@ -3,51 +3,54 @@ jobs:
...
+
+      # Build the skeleton
       - run:
-          name: Build packages
+          name: Build modern BPF skeleton 🐝
           command: |
-            docker run -e BUILD_TYPE="release" -it -v /tmp/source-arm64:/source -v /tmp/build-arm64:/build \
-              falcosecurity/falco-builder:latest \
-              package
+            mkdir -p /tmp/source-arm64/falco/skeleton-build
```
Comment (Andreagit97) [inline .circleci/config.yml:31]: here we build in a separate build dir called `skeleton-build` the modern bpf skeleton

File: .circleci/config.yml [hunk 4]
```diff
@@ -3,51 +3,54 @@ jobs:
...
+      # - the 3 packages: tar, deb, rpm. (under /tmp/packages)
       - run:
-          name: Prepare Artifacts
+          name: Build Falco packages 🏗️
           command: |
-            mkdir -p /tmp/packages
-            cp /tmp/build-arm64/release/*.deb /tmp/packages
-            cp /tmp/build-arm64/release/*.tar.gz /tmp/packages
-            cp /tmp/build-arm64/release/*.rpm /tmp/packages
+            DOCKER_BUILDKIT=1 docker build -f /tmp/source-arm64/falco/docker/builder/modern-falco-builder.Dockerfile --output type=local,dest=/tmp --build-arg CMAKE_OPTIONS="-DCMAKE_BUILD_TYPE=Release -DUSE_BUNDLED_DEPS=On -DFALCO_ETC_DIR=/etc/falco -DBUILD_FALCO_MODERN_BPF=ON -DMODERN_BPF_SKEL_DIR=/source/skeleton-build/skel_dir -DBUILD_DRIVER=Off -DBUILD_BPF=Off" --build-arg DEST_BUILD_DIR=/build-arm64/release /tmp/source-arm64/falco
```
Comment (Andreagit97) [inline .circleci/config.yml:42]: * `DOCKER_BUILDKIT` is needed to use the `--output` docker command
* as a build arg we pass `CMAKE_OPTIONS=`, so all the CMake options necessary to build Falco with the modern probe, note that we need to provide the external built skeleton with the `MODERN_BPF_SKEL_DIR` CMake variable, note that the provided path is already relative to the container file system not to the host one.
* `--build-arg DEST_BUILD_DIR=/build-arm64/release` here we are saying the container to copy the build directory in this location of the ubuntu 22.04 machine, in this way we can reuse this build directory in the `test-integrations`
* `/tmp/source-arm64/falco` this is the docker build context with the source code of Falco

**Review by Andreagit97 — COMMENTED** [review]
Added some comments on arm64 job, they are all valid for the centos7 one

File: test/CMakeLists.txt [hunk 5]
```diff
@@ -1,4 +1,6 @@
 add_subdirectory(trace_files)
 
-add_subdirectory(plugins)
-add_subdirectory(confs/plugins)
+if(NOT MUSL_OPTIMIZED_BUILD)
```
Comment (Andreagit97) [inline test/CMakeLists.txt:3]: in musl build we don't have plugins so I've disabled the testing phase for plugins

**Comment by Andreagit97** [comment]
I've bumped the resource classes to `large` in `build-centos7` and `build-arm64` jobs since we noticed some timeout issues in these 2 jobs.

**Comment by FedeDP** [comment]
This PR also bumps libs and driver to latest tags: 0.10.0 and 4.0.0+driver! Thanks Andrea!

**Review by FedeDP — DISMISSED** [review]
/approve

**Review by FedeDP — APPROVED** [review]
/approve

**Review by jasondellaluce — APPROVED** [review]
/approve

**Pull request merged**